security/oc_tls.c

```diff
@@ -390,12 +403,39 @@ oc_tls_free_peer(oc_tls_peer_t *peer, bool inactivity_cb)
     oc_message_unref(message);
     message = (oc_message_t *)oc_list_pop(peer->recv_q);
   }
+  oc_process_drop(&oc_tls_handler, process_drop_event_for_removed_endpoint,
+                  &peer->endpoint);
 #ifdef OC_PKI
   oc_free_string(&peer->public_key);
 #endif /* OC_PKI */
   mbedtls_ssl_config_free(&peer->ssl_conf);
   oc_etimer_stop(&peer->timer.fin_timer);
+
+  oc_endpoint_t endpoint;
```

Comment:
Why is the local copy needed? It gets deallocated when the function ends, same way the peer::endpoint was previously. What is different?

Reply:
I moved deallocation before the call callbacks. Peer is a big struct so in constrain device you want to free resource. Eg in one of the callbacks there could be created new peer